qiime2/sdk/tests/test_usage.py

```diff
@@ -359,4 +363,29 @@ def test_merge_metadata_one_input(self):
 class TestScopeRecord(TestCaseUsage):
     def test_invalid_assert_has_line_matching(self):
         with self.assertRaisesRegex(TypeError, 'should be a `callable`'):
-            usage.ScopeRecord('foo', assert_has_line_matching='spleen')
+            usage.ScopeRecord('foo', 'value', 'source',
+                              assert_has_line_matching='spleen')
+
+
+class TestExecutionUsage(TestCaseUsage):
+    def test_init_data(self):
+        use = usage.ExecutionUsage()
+
+        with self.assertRaisesRegex(ValueError, 'expected an Artifact'):
+            use.init_data('name', lambda: object)
+
+        with self.assertRaisesRegex(ValueError, 'not all .* Artifacts'):
+            use.init_data('name', lambda: [object])
+
+        with self.assertRaisesRegex(TypeError, 'must be of type Metadata'):
+            use.init_metadata('name', lambda: object)
+
+        with self.assertRaisesRegex(ValueError, '`ref` must match'):
+            use.get_metadata_column('name', Metadata, 'a')
```

Comment:
Is there a reason not to change the signature of `Usage.get_metadata_column` to something like `get_metadata_column(self, column_name, record)` and pass that through as the `ref`?

Reply:
Great idea - let's do that.